File: widgetssdk/src/main/java/com/glia/widgets/filepreview/data/GliaFileRepository.java
package com.glia.widgets.filepreview.data;

import android.graphics.Bitmap;

import com.glia.androidsdk.chat.AttachmentFile;

import io.reactivex.Completable;
import io.reactivex.Maybe;

public interface GliaFileRepository {
    Maybe<Bitmap> loadImageFromCache(String fileName);

    Completable putImageToCache(String fileName, Bitmap bitmap);

    Maybe<Bitmap> loadImageFromDownloads(String fileName);

    Completable putImageToDownloads(String fileName, Bitmap bitmap);

    Maybe<Bitmap> loadImageFileFromNetwork(AttachmentFile file);

    Completable downloadFileFromNetwork(AttachmentFile file);
}

File: widgetssdk/src/main/java/com/glia/widgets/filepreview/data/GliaFileRepository.kt
package com.glia.widgets.filepreview.data

import android.graphics.Bitmap
import com.glia.androidsdk.chat.AttachmentFile
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Maybe
import java.io.InputStream

internal interface GliaFileRepository {
    fun isReadyForPreview(attachmentFile: AttachmentFile): Boolean
    fun loadImageFromCache(fileName: String): Maybe<Bitmap>
    fun putImageToCache(fileName: String, bitmap: Bitmap)
    fun loadImageFromDownloads(fileName: String): Maybe<Bitmap>
    fun putImageToDownloads(fileName: String, bitmap: Bitmap): Completable
    fun loadImageFileFromNetwork(attachmentFile: AttachmentFile): Maybe<InputStream>
    fun downloadFileFromNetwork(attachmentFile: AttachmentFile): Completable
}
